lib/js/dom.nim

```diff
@@ -1511,10 +1515,13 @@ proc routeEvent*(w: Window, event: Event)
 proc scrollBy*(w: Window, x, y: int)
 proc scrollTo*(w: Window, x, y: int)
 proc setInterval*(w: Window, code: cstring, pause: int): ref Interval
+proc setInterval*[T](w: Window, function: proc (): T, pause: int): ref Interval
 proc setInterval*(w: Window, function: proc (), pause: int): ref Interval
 proc setTimeout*(w: Window, code: cstring, pause: int): ref TimeOut
+proc setTimeout*[T](w: Window, function: proc (): T, pause: int): ref Interval
 proc setTimeout*(w: Window, function: proc (), pause: int): ref Interval
 proc stop*(w: Window)
+proc requestAnimationFrame*[T](w: Window, function: proc (time: float): T): int
```

Comment:
`time` should be `SomeFloat` IMHO.

Reply:
same comment as https://github.com/nim-lang/Nim/pull/17406#discussion_r598048100 , let's wait until this is implemented